src/render.tsx

```diff
@@ -57,6 +76,35 @@ export default function render<T>(
   return result;
 }
 
+export default function renderComponent<T>(
+  component: React.ReactElement<T>,
+  { wrapper: Wrapper, createNodeMock, validateRenderedStrings }: Options = {}
+) {
+  if (!validateRenderedStrings) {
+    return render({ component, wrapper: Wrapper, createNodeMock });
+  }
+
+  const handleRender: React.ProfilerProps['onRender'] = (_, phase) => {
+    if (phase === 'update') {
+      assertStringsWithinText(screen.container);
+    }
+  };
+
+  const wrap = (innerElement: React.ReactElement) => (
+    <Profiler id="renderProfiler" onRender={handleRender}>
```

Comment:
❗️ @pierrezimmermannbam 
We would want to avoid wrapping in `<Profiler>` if the user does not enable this option. So for regular renders the behaviour stays exactly the same.

Reply:
If the option is not provided we return render without wrapping with a Profiler, we have a check on the option both in the renderComponent function (for component updates) and in the render function (for check after mount)